qiita_db/meta_util.py

```diff
@@ -163,3 +168,34 @@ def get_accessible_filepath_ids(user):
             filepath_ids.update(analysis.all_associated_filepath_ids)
 
         return filepath_ids
+
+
+def get_lat_longs():
+    """Retrieve the latitude and longitude of all the samples in the DB
+
+    Returns
+    -------
+    list of [float, float]
+        The latitude and longitude for each sample in the database
+    """
+    portal_table_ids = Portal(qiita_config.portal).get_studies()
+
+    with TRN:
+        sql = """SELECT DISTINCT table_name
+                 FROM information_schema.columns
+                 WHERE SUBSTR(table_name, 1, 7) = 'sample_'
+                    AND table_schema = 'qiita'
+                    AND column_name IN ('latitude', 'longitude');"""
+        TRN.add(sql)
+
+        sql = "SELECT latitude, longitude FROM qiita.{0}"
+        idx = TRN.index
+
+        portal_tables = TRN.execute_fetchflatten()
+        portal_tables = [x for x in TRN.execute_fetchflatten()
```

Comment:
I though that avoiding the python loop will be faster, as is one less loop. Using the live system, both take the same time:
```
In [33]: %timeit old_get_lat_longs()
1 loops, best of 3: 1.24 s per loop

In [34]: %timeit new_get_lat_longs()
1 loops, best of 3: 1.24 s per loop
```

I think the main problem is the for loop in line 198... Feel free to not change it if you don't feel like.